Automattic/wp-calypso
# Signup: Add free plan + paid domain modal

Fixes Automattic/martech#1649

## Proposed Changes
The full spec is in Automattic/martech#1649.

#### Domains Step
* Change in the free domain explainer copy:
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/5436027/231076707-c5d64817-fa15-4d4b-9ce1-c989c4a409fc.png">

#### Plans Step
* Change in the domain feature copy for paid plans:
<img width="499" alt="image" src="https://user-images.githubusercontent.com/5436027/231077206-d25917e2-606c-4e60-882e-7e64ece45eb9.png">

* Change in the domain feature copy for the free plan. The `.wordpress.com` subdomain is fetched via a new hook `useGetWordPressSubdomain`.
<img width="283" alt="image" src="https://user-images.githubusercontent.com/5436027/231077480-90faa318-4e97-40df-a199-8e19eb221039.png">

* The free plan + paid domain modal when a paid domain has been selected, and the user clicks on the Free plan CTA. This uses the same hook, and the value is cached for 5 minutes, so in most cases, a separate network is not required to fetch the subdomain.
<img width="1679" alt="image" src="https://user-images.githubusercontent.com/5436027/231676748-7b5784df-0070-480a-903d-a19a61ce077d.png">



## Testing Instructions



1. Go to `/start`.
1. On the domains step, search for some domains. Confirm that the free-domain explainer text (in green) matches the spec.
1. Select a paid domain.
1. On the plans step, confirm that the domain feature for paid plans matches the spec.
1. Confirm that a `wordpress.com` subdomain is displayed under the crossed-out domain name in the feature list of the free plan.
1. Click on the `Start with Free` CTA.
1. Confirm that the modal is shown, and the subdomain matches the one shown on the plans grid.
1. Confirm that the price shown for the Personal plan is correct.
1. Select `Get Personal`, and confirm that you are taken to the checkout page with the selected paid domain and the Personal plan in the cart.
1. Repeat steps 1-8.
1. Select "Continue with Free".
1. Confirm that a free site is created with free subdomain shown on the modal. 

## Pre-merge Checklist
